benchmark/net/chunky_http_client.js

```diff
@@ -0,0 +1,111 @@
+// test HTTP throughput in fragmented header case
+var common = require('../common.js');
+var server;
+var test = require('../../test/common.js');
+var PORT = common.PORT;
+var PIPE = test.PIPE;
+
+var net = require('net');
+var bench = common.createBenchmark(main, {
+  len: [1, 4, 8, 16, 32, 64, 128],
+  num: [5, 50, 500, 2000],
+  type: ['send'],
+}
+)
+
+var len;
+var num;
+var type;
+
+function main(conf) {
+  num = +conf.num;
+  len = +conf.len;
+  type = conf.type;
+  var todo = [ ];
+  var headers = [ ];
+  for (var i = 0; i < 3; i += 1) {
+    var c = Math.pow(7, i+1)
+    var str = '';
+    for (var j = 0; j < c; ++j) {
+      str += 'o';
+    }
+    headers[i] = str;
+  }
+
+  function WriteWithCRLF(line) {
+    todo.push(line + '\r\n');
+  }
+
+  function WriteHTTPHeaders(channel, has_keep_alive, extra_header_count) {
+    todo = [ ]
+    WriteWithCRLF('GET / HTTP/1.1');
+    WriteWithCRLF( 'Host: localhost');
+    if (has_keep_alive) {
+      WriteWithCRLF( 'Connection: keep-alive');
+    }
+    WriteWithCRLF( 'Accept: text/html,application/xhtml+xml,' +
+                           'application/xml;q=0.9,image/webp,*/*;q=0.8');
+    WriteWithCRLF( 'User-Agent: Mozilla/5.0 (X11; Linux x86_64) ' +
+                           'AppleWebKit/537.36 (KHTML, like Gecko) ' +
+                           'Chrome/39.0.2171.71 Safari/537.36');
+    WriteWithCRLF( 'Accept-Encoding: gzip, deflate, sdch');
+    WriteWithCRLF( 'Accept-Language: en-US,en;q=0.8');
+    var i;
+    for (i = 0; i < extra_header_count; i++) {
```

Comment:
Just write this as `for (var i = ...)`, the vardecl gets hoisted to the function scope anyway.